liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppsMojo.java

```diff
@@ -80,9 +117,72 @@ protected void doExecute() throws Exception {
             log.warn(messages.getString("warn.install.app.add.configuration"));
             applicationXml.writeApplicationXmlDocument(serverDirectory);
         }
+    }
+
+    private void installSpringBootApp() throws Exception {
+        File fatArchiveSrc = getFatArchiveSrc();
+        //Check if the archiveSrc is executable and then invokeSpringUtilCommand. 
+        if(isFileExecutable(fatArchiveSrc)) {
+            File thinArchiveTarget = getThinArchiveTarget(fatArchiveSrc);
+            File libIndexCacheTarget = getLibIndexCacheTarget();
+            
+            Artifact artifact = project.getArtifact();
+            artifact.setFile(thinArchiveTarget);
+            
+            validateAppConfig(artifact.getFile().getName(), artifact.getArtifactId(), true);
+            invokeSpringBootUtilCommand(installDirectory, fatArchiveSrc.getCanonicalPath(), thinArchiveTarget.getCanonicalPath(), libIndexCacheTarget.getCanonicalPath());
+        } else {
+            throw new MojoExecutionException(fatArchiveSrc.getCanonicalPath() +" file is not an executable archive. "
+                    + "The repackage goal of the spring-boot-maven-plugin must be configured to run first in order to create the required executable archive.");
+        }
+    }
+    
+    private File getFatArchiveSrc() {
+        String classifier = (this.classifier == null ? "" : this.classifier.trim());
+        if (!classifier.isEmpty() && !classifier.startsWith("-")) {
+            classifier = "-" + classifier;
+        }
+        if (!this.outputDirectory.exists()) {
+            this.outputDirectory.mkdirs();
+        }
+        return new File(this.outputDirectory, this.finalName + classifier + "." + getArtifactExtension());
+    }
+    
+    private File getThinArchiveTarget(File archiveSrc) {
+        String appsDirName = getAppsDirectory();
+        File archiveTarget = null;
+        File appsDir = null;
+        
+        if("apps".equals(appsDirName)){
+            appsDir = new File(serverDirectory, appsDirName);        
+        } else if ("dropins".equals(appsDirName)) {
+            appsDir = new File(serverDirectory, appsDirName+"/spring");         
+        }       
+        archiveTarget = new File(appsDir, "thin-" + archiveSrc.getName());
+        return archiveTarget;
+    }
+    
+    private File getLibIndexCacheTarget() {
+        File libIndexCacheTarget = new File(sharedResourcesDirectory, "lib.index.cache");
+        return libIndexCacheTarget;
+    }
+ 
+    @SuppressWarnings("resource")
+    private boolean isFileExecutable(File archiveSrc) throws IOException {
+        if(archiveSrc.exists()) {
+            String startClass = new JarFile(archiveSrc).getManifest().getMainAttributes().getValue("Start-Class");
```

Comment:
The getManifest() method could potentially return null which would cause a NPE here.